syntax = "proto3";
package gRPCServices;
import "pose.proto";
import "google/protobuf/empty.proto";

message AttachMessage {
enum Device {
    UNSPECIFIED = 0;
    HMD = 1;
    TRACKER = 2;
    CONTROLLER = 3;
}
  int32 id = 1;
  Device device = 2;
}
message DetachMessage {
  int32 id = 1;
}
message TransformMessage {
  int32 id = 1;
  oneof relative_absolute_oneof{
    Graphics.CartesianPose absolute = 2;
    Graphics.CartesianPose relative = 3;
  }
}
service InputEmulation {
rpc Attach(AttachMessage) returns (google.protobuf.Empty);
rpc Detach(DetachMessage) returns (google.protobuf.Empty);
rpc Transform(TransformMessage) returns (google.protobuf.Empty);
}